### PR TITLE
Allow named structs to be embedded

### LINKED ISSUE #277: Add support for embedded no anonymous fields
Change

```go
// struct without primary key is a field
// TODO: test by scanner/valuer instead?
if pk, _ := searchPrimary(typ); len(pk) == 0 {
	data.fields = append(data.fields, name)
	continue
}
```

If struct implement `scanner/valuer` - is field, else is embedded

### PR DESCRIPTION
**Pull request by dranikpg**
Follow up on #262 that fixes #277 and #296

### CODE REVIEW

File: document_meta.go [hunk 1]
```diff
@@ -357,6 +357,21 @@ func fieldName(sf reflect.StructField) (string, bool) {
 	return snaker.CamelToSnake(sf.Name), false
 }
 
+func isEmbedded(sf reflect.StructField) bool {
+	// anonymous structs are always embedded
```
Comment (dranikpg) [inline document_meta.go:361]: Is this still true to provide backwards compatibility?

Comment (Fs02) [reply document_meta.go:361]: anonymous struct is basically an embedded struct, so I think no problem

File: document_meta.go [hunk 2]
```diff
@@ -357,6 +357,21 @@ func fieldName(sf reflect.StructField) (string, bool) {
...
+	if sf.Anonymous {
+		return true
+	}
+	// search for embedded tag
+	tags := strings.Split(sf.Tag.Get("db"), ",")
+	for i, tag := range tags {
+		if tag == "embedded" && i > 0 {
```
Comment (Fs02) [inline document_meta.go:368]: maybe instead of `i > 0`, better to loop by index starting 1? 🤔

Comment (dranikpg) [reply document_meta.go:368]: To not accidentally step out of bounds

File: document_meta.go [hunk 3]
```diff
@@ -357,6 +357,21 @@ func fieldName(sf reflect.StructField) (string, bool) {
...
 }
 
+func isEmbedded(sf reflect.StructField) bool {
+	// anonymous structs are always embedded
+	if sf.Anonymous {
+		return true
+	}
+	// search for embedded tag
+	tags := strings.Split(sf.Tag.Get("db"), ",")
+	for i, tag := range tags {
```
[the remark this answers is not in the document]
Comment (dranikpg) [reply document_meta.go:367]: If this is fine, let's use it. Just feels a bit weird as this means that we can't have more than one struct tag (except for the name)

**Review by Fs02 — APPROVED** [review]
LGTM 👾

**Pull request merged**